### PR TITLE
Started on cons_canvas color support.

### PR DESCRIPTION
**Pull request by r0nk**
(incomplete)

Started on r_cons_canvas color support, in order to allow colored graph edges. Still working out the bugs and strange behavior.

### CODE REVIEW

File: libr/cons/canvas.c [hunk 1]
```diff
@@ -34,6 +49,14 @@ R_API RConsCanvas* r_cons_canvas_new (int w, int h) {
 		free (c);
 		return NULL;
 	}
+	c->attrslen = 0;
+	c->attrs = calloc(sizeof(*c->attrs),c->blen+1);
+	if (!c->attrs) {
+		if (c->b)
```
Comment (jvoisin) [inline libr/cons/canvas.c]: There is no need to check if `c->b` is NULL before calling `free` on it.

**Comment by r0nk** [comment]
Aight, I Worked out a few more bugs in the color system, and I've pulled the old algorithm I used on my debugger for drawing lines. Let me know what you think about the lines.

**Comment by crowell** [comment]
color is still broken
see

**Comment by r0nk** [comment]
Woops yeah I meant to say that, Its still not complete yet, the colors are being worked on.

**Comment by radare** [comment]
Good work! Looking forward your complete patch to merge

**Comment by r0nk** [comment]
Colors are now working, but they make Ricky bobby cry. I'll see about profiling r_cons_canvas and optimizing it.

**Comment by crowell** [comment]
stil colors look broken.

using konsole, with this tmux config http://raxcity.com/.tmux.conf in 256 colors mode, this is super broken too.

**Pull request closed without merging**